### PR TITLE
pythonPackages.sqlite-utils: depend on sqlite-fts4

### PR DESCRIPTION
**Pull request by meatcar**
###### Motivation for this change

Adding `sqlite-fts4` as a dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104253` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.sqlite-fts4</li>
    <li>python37Packages.sqlite-utils</li>
    <li>python38Packages.sqlite-fts4</li>
    <li>sqlite-utils (python38Packages.sqlite-utils)</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104253` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.sqlite-fts4</li>
    <li>python37Packages.sqlite-utils</li>
    <li>python38Packages.sqlite-fts4</li>
    <li>sqlite-utils (python38Packages.sqlite-utils)</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104253` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.sqlite-fts4</li>
    <li>python37Packages.sqlite-utils</li>
    <li>python38Packages.sqlite-fts4</li>
    <li>sqlite-utils (python38Packages.sqlite-utils)</li>
  </ul>
</details>

File: pkgs/development/python-modules/sqlite-fts4/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    pytestCheckHook
+  ];
+
+  meta = with lib; {
+    description = "Custom Python functions for working with SQLite FTS4";
+    homepage = "https://github.com/simonw/sqlite-fts4";
+    license = licenses.asl20;
+    maintainers = [ maintainers.meatcar ];
+  };
+
```
[the remark this answers is not in the document]
Comment (meatcar) [reply pkgs/development/python-modules/sqlite-fts4/default.nix]: Thanks, change implemented for both packages in the PR.

**Comment by meatcar** [comment]
I see there was a merge conflict due to PR#105368 forcing the update to `sqlite-utils` 3.0 in b8c648 without adding the `sqlite-fts4` dependency. 

I've resolved the conflict, but for some reason there are two failing tests across all python versions, I'm investigating.

**Comment by jonringer** [comment]
looks to be a small flakey difference. I'm okay with just disabling the tests

```
builder for '/nix/store/x10pnlagpnchjc73wqp38isjq1045pzz-python3.7-sqlite-utils-3.0.drv' failed with exit code 1; last 10 log lines:
          assert 0 == result.exit_code
          size_after_optimize = os.stat(db_path).st_size
  >       assert size_after_optimize < size_before_optimize
  E       assert 1667072 < 1662976
```

**Review by vikanezrimaya — COMMENTED** [review]
Ok, this pull-request is a little bit confusing. First of all, do I understand correctly that there is already a commit bumping `sqlite-utils` to 3.0 but forgetting to add a dependency? how was this missed before?! :scream_cat:

The `sqlite-fts4` builds fine. I don't think this PR should break anything (it's not removing dependencies, only adding them, only side-effect could be closure growth), but there was a mention of a flaky test, so I can't give it an LGTM but I don't feel like I'm qualified to request changes here, since it's so confusing ~~and the tests just plain hung up on my machine so I'm not sure what's happening~~ and the tests did fail for me, and they apparently take some time to perform.

**Comment by meatcar** [comment]
Thanks for taking a look, I agree this PR is pretty confusing at the moment. I was just going to poke it awake earlier today when I saw the merge conflict, and had to run.

Is there a better way I can clarify the confusion through the commit history? Should I revert b8c6488 instead, to keep the changes together and more atomic?

I have a hunch on how to fix the tests, I'll test it as soon as I get back to my pc.

**Comment by kisik21** [comment]
I don't think the faulty commit should be reverted, rather this PR should be renamed to better represent that a dependency is added. The commit message should acknowledge the faulty commit and point out the mistake that was left there - and that now it is fixed.

**Comment by meatcar** [comment]
Thanks @kisik21, I've reworded the commit.

I've also tracked down the cause for the failing tests, its due to an update of sqlite from 3.33 to 3.34. The tests fail at cfc56fb90bed519da166a9d440204b0dd7a6b6aa, and pass at the immediately previous e4d17dc558f (after waiting for 150 packages to rebuild overnight). I've opened an issue upstream (https://github.com/simonw/sqlite-utils/issues/209), and disabled the failing two tests for now.

Testing this tool is straight forward:
```
$ cp /nix/var/nix/db/db.sqlite /tmp/db.sqlite
$ ./result/bin/sqlite-utils enable-fts --fts4 /tmp/db.sqlite ValidPaths path
$ ./result/bin/sqlite-utils search /tmp/db.sqlite ValidPaths firefox
```

Also `optimize` still works, I think it only fails for the contrived example in the tests because the entire DB is set up for full-text searching.
```
$ find /tmp/db.sqlite -printf '%s'
46256128
$ ./result/bin/sqlite-utils optimize /tmp/db.sqlite
$ find /tmp/db.sqlite -printf '%s'
30429184
```

**Comment by kisik21** [comment]
I've run your example and I can confirm that this is working. Should we merge this now or wait for an upstream to fix the failing test? I'll leave this decision to whoever will be assigned to merge this, since I'm not yet well-versed in Nixpkgs policies, but if I were to decide, I'd err on the side of merging, as the functionality seems to be working as intended (as far as demonstrated by the remaining enabled tests and the manual test case described here).

/status needs_merger

**Pull request merged**